### PR TITLE
Add deploy keys for Slurm appliance

### PR DESCRIPTION
**Pull request by mkjpryor**

### CODE REVIEW

File: azimuth_caas_operator/utils/ansible_runner.py [hunk 1]
```diff
@@ -26,31 +108,19 @@ def get_env_configmap(
     extraVars["cluster_id"] = cluster.metadata.uid
     extraVars["cluster_type"] = cluster.spec.clusterTypeName
     extraVars["cluster_deploy_ssh_public_key"] = cluster_deploy_ssh_public_key
-    extraVars["cluster_ssh_private_key_file"] = "/home/runner/.ssh/id_rsa"
-
+    # This is the file containing the private key for the deploy key
+    extraVars["cluster_ssh_private_key_file"] = "/var/lib/caas/ssh/id_ed25519"
```
Comment (JohnGarbutt) [inline azimuth_caas_operator/utils/ansible_runner.py:112]: This will break graphcore's appliance, why can't we keep the older path?

**Comment by JohnGarbutt** [comment]
I am worried by the functional tests breaking here, do you know why?

**Comment by mkjpryor** [comment]
Yeah - I didn’t like those defaults as they almost never apply 🤣 Better to be explicit I think.

File: azimuth_caas_operator/utils/ansible_runner.py [hunk 2]
```diff
@@ -364,8 +456,8 @@ async def _get_ansible_runner_events(client, job_name, namespace):
     if len(pod_names) == 0 or len(pod_names) > 1:
         # TODO(johngarbutt) only works because our jobs don't retry,
         # and we don't yet check the pod is running or finished
-        LOG.debug(f"Found pods:{pod_names} for job {job_name} in {namespace}")
-        return
+        LOG.debug(f"Found pods: {pod_names} for job {job_name} in {namespace}")
+        return []
```
Comment (JohnGarbutt) [inline azimuth_caas_operator/utils/ansible_runner.py:460]: NIT: this should really be a separate PR with a nice commit message about what if fixes.

**Review by JohnGarbutt — CHANGES_REQUESTED** [review]
OK, this looks nice.

Three things I think we should change though:
1. Lets not default to the latest tag for the ee docker image, just require that environment variable like with consul
2. There are two bug fixes in there that should be in separate PRs please, so we have the proper git history around those fixes, the return [] one and the untag job one.

File: azimuth_caas_operator/operator.py [hunk 3]
```diff
@@ -253,6 +253,7 @@ async def cluster_update(body, name, namespace, labels, **kwargs):
             if reason:
                 error += f" Possible reason for the failure was: {reason}"
 
+            await ansible_runner.unlabel_job(K8S_CLIENT, update_job)
```
[the remark this answers is not in the document]
Comment (mkjpryor) [reply azimuth_caas_operator/operator.py]: Done

File: azimuth_caas_operator/utils/ansible_runner.py (hunk 2 above)
Comment (mkjpryor) [reply azimuth_caas_operator/utils/ansible_runner.py:460]: Done

File: azimuth_caas_operator/utils/image.py [hunk 4]
```diff
@@ -1,7 +1,18 @@
 import os
 
-DEFAULT_ANSIBLE_RUNNER_IMAGE = "ghcr.io/stackhpc/azimuth-caas-operator-ar:v0.1.0"
+
+DEFAULT_ANSIBLE_RUNNER_IMAGE_REPO = "ghcr.io/stackhpc/azimuth-caas-operator-ee"
+DEFAULT_ANSIBLE_RUNNER_IMAGE_TAG = "latest"
```
[the remark this answers is not in the document]
Comment (mkjpryor) [reply azimuth_caas_operator/utils/image.py:5]: Done

File: azimuth_caas_operator/utils/image.py [hunk 5]
```diff
@@ -1,7 +1,18 @@
...
+    if "ANSIBLE_RUNNER_IMAGE" in os.environ:
+        return os.environ["ANSIBLE_RUNNER_IMAGE"]
+    else:
+        repo = os.environ.get(
+            "ANSIBLE_RUNNER_IMAGE_REPO", DEFAULT_ANSIBLE_RUNNER_IMAGE_REPO
+        )
+        try:
+            tag = os.environ["ANSIBLE_RUNNER_IMAGE_TAG"]
+        except KeyError:
+            raise RuntimeError("ANSIBLE_RUNNER_IMAGE_TAG is not set")
```
Comment (JohnGarbutt) [inline azimuth_caas_operator/utils/image.py:17]: cool

**Review by JohnGarbutt — APPROVED** [review]
Looks good now, thank you!
